src/auth/auth.service.ts

```diff
@@ -6,24 +6,35 @@ import * as bcrypt from 'bcrypt';
 import { JwtService } from '@nestjs/jwt';
 import { CreateUserDto } from 'src/users/dto/create-user.dto';
 import { SanitizedUser } from 'src/users/types/sanitized-user.type';
+import { MailService } from 'src/mail/mail.service';
+import { PasswordRequestDto } from './dto/password-request.dto';
+import {v4 as uuidv4} from 'uuid';
+import { PasswordResetDto } from './dto/password-reset.dto';
 
 @Injectable()
 export class AuthService {
   constructor(
     private usersService: UsersService,
-    private jwtService: JwtService
+    private jwtService: JwtService,
+    private mailService: MailService,
   ) {}
 
   async register(createUserDto: CreateUserDto): Promise<SanitizedUser> {
     if (!await this.usersService.validateEmail(createUserDto.email)) {
-        throw new HttpException("Registration error - this email is already used", HttpStatus.BAD_REQUEST);
+      throw new HttpException("Registration error - this email is already used", HttpStatus.BAD_REQUEST);
     }
 
     if (!await this.usersService.validateUsername(createUserDto.username)) {
-        throw new HttpException("Registration error - this username is already used", HttpStatus.BAD_REQUEST);
+      throw new HttpException("Registration error - this username is already used", HttpStatus.BAD_REQUEST);
     }
 
-    return await this.usersService.create(createUserDto);
+    let user = await this.usersService.create(createUserDto);
+    
+    if (!await this.mailService.sendUserConfirmation(createUserDto.email, user.emailNonce)) {
```

Comment:
Make use of precedently generated `user`! It comes from the database, which is the source of truth. DTOs should only be used as long as the data is in transit towards the database.

```suggestion
    if (!await this.mailService.sendUserConfirmation(user.email, user.emailNonce)) {
```